lib/kapost_deploy/task.rb

```diff
@@ -87,21 +87,28 @@ def define
 
     private
 
-    def initialize(name)
+    def initialize(name, scope: Rake.application.current_scope.path)
       defaults
-      self.name = name
+      @name = name
+      @scope = scope
     end
 
     attr_accessor :plugins
+    attr_reader :scope
+
+    def hook_name(type)
+      return "#{name}:#{type}_#{name}" if scope.empty?
+      "#{scope}:#{name}:#{type}_#{name}"
```

Comment:
Thanks, I had considered that but the problem is that when the namespace is missing, it answers an empty string instead so `#compact` won't catch it.

I have a [refinement](https://github.com/bkuhlmann/refinements#array) for this if we wanted to add a gem dependency that would make this more elegant if you're up for it?